packages/web/src/ui/views/viewer.js

```diff
@@ -117,10 +108,34 @@ const viewer = (state, i18n) => {
         controls = { ...controls, ...updated.controls }
       })
 
+    const doRotatePanZoom = (timestamp) => {
+      timer = null;
+      if(delta[0] || delta[1]){      
+        const updated = orbitControls.rotate({ controls, camera, speed: rotateSpeed }, delta)
+        delta = [0,0];
+        controls = { ...controls, ...updated.controls }
+      }
+      
+      if(deltaPan[0] || deltaPan[1]){      
+        const updated = orbitControls.pan({ controls, camera, speed: panSpeed }, deltaPan)
+        deltaPan = [0,0];
+        camera.position = updated.camera.position
+        camera.target = updated.camera.target
+      }
+
+      if(zoomDelta){
+        const updated = orbitControls.zoom({ controls, camera, speed: zoomSpeed }, zoomDelta)
+        controls = { ...controls, ...updated.controls }
+        zoomDelta = 0;
+      }
+
+      updateAndRender(timestamp, true)
```

Comment:
Afer looking at the code, I think adding updateAndRender() here was a mistake. Sorry.
This could potentially create several concurrent requestAnimationFrame() loops, which would definitely effect performance.

Reply:
Yeas, I was trying out ideas to see how the interaction feels, to make it more responsive to mouse inputs.